src/In.ProjectEKA.TMHHip/DataFlow/Collect.cs

```diff
@@ -0,0 +1,550 @@
+                            new Note(swellingSymptomData.SwellingSize)
+                        }),
+                }).Cast<ICondition>().ToList();
+
+            if (!conditions.Any())
+            {
+                return Option.None<ConditionResponse>();
+            }
+
+            var conditionResponse = new ConditionResponse
+            {
+                Entry = conditions,
+                Id = Uuid.Generate().Value.Split(":").Last(),
+                ResourceType = "Bundle",
+                Type = "collection"
+            };
+            return Option.Some(conditionResponse);
+        }
+
+        private static async Task<Option<ObservationResponse>> FetchAllergiesData(DataRequest dataRequest,
+            List<AllergyData> allergiesData, string patientName)
+        {
+            LogDataRequest(dataRequest);
+            var uuid = Uuid.Generate().Value;
+            var id = uuid.Split(":").Last();
+            var observations = (from allergyData in allergiesData
+                where WithinRange(dataRequest.DateRange, allergyData.AllergyDate)
+                select new ObservationRepresentation
+                {
+                    FullUrl = uuid,
+                    Resource = new Resource(HiType.Observation, id, "final", new Code("Allergies"),
+                        new Subject(patientName), allergyData.AllergyDate, allergyData.Allergies,
+                        new List<Note> {new Note(allergyData.AllergyRemark)}),
+                }).ToList();
+
+            if (!observations.Any()) return Option.None<ObservationResponse>();
+            var observationResponse = new ObservationResponse
+            {
+                Entry = observations,
+                Id = Uuid.Generate().Value.Split(":").Last(),
+                ResourceType = "Bundle",
+                Type = "collection"
+            };
+            return Option.Some(observationResponse);
+        }
+
+        private static async Task<Option<ObservationResponse>> FetchAbdomenExaminationData(DataRequest dataRequest,
+            List<AbdomenExaminationData> abdomenExaminationsData, string patientName)
+        {
+            LogDataRequest(dataRequest);
+            var uuid = Uuid.Generate().Value;
+            var id = uuid.Split(":").Last();
+            var observations = (from abdomenExaminationData in abdomenExaminationsData
+                where WithinRange(dataRequest.DateRange, abdomenExaminationData.EffectiveDateTime)
+                select new ObservationRepresentation
+                {
+                    FullUrl = uuid,
+                    Resource = new Resource(HiType.Observation, id, "final", new Code("Examination notes for Abdomen"),
+                        new Subject(patientName), abdomenExaminationData.EffectiveDateTime,
+                        abdomenExaminationData.CAbdomen),
+                }).ToList();
+
+            if (!observations.Any()) return Option.None<ObservationResponse>();
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4940156)